cmd/controller/main.go

```diff
@@ -84,7 +112,7 @@ func main() {
 	ctx = restconfig.Inject(ctx, config)
 
 	// 3. Set up controller runtime controller
-	cloudProvider := registry.NewCloudProvider(ctx, cloudprovider.Options{ClientSet: clientSet})
+	cloudProvider := registry.NewCloudProvider(ctx, cloudprovider.Options{ClientSet: clientSet, ClusterName: options.ClusterName, ClusterEndpoint: options.ClusterEndpoint})
```

Comment:
What do you think about using the context pattern to store the options? 
```ctx = options.Inject(ctx, config)```
Then we can pull any flags we need out anywhere in the code.

Reply:
That's not a bad idea. I'm not a huge fan of storing cli args in `context`, but it might make sense.  I'll try it out and see how it feels